autoload/ale/fixers/yapf.vim

```diff
@@ -1,13 +1,18 @@
 " Author: w0rp <devw0rp@gmail.com>
 " Description: Fixing Python files with yapf.
 
-function! ale#fixers#yapf#Fix(buffer) abort
+function! ale#fixers#yapf#Fix(buffer, lines) abort
+    let l:executable = ale#handlers#python#GetExecutable(a:buffer, 'yapf')
+    if empty(l:executable)
+        return a:lines
+    endif
+
     let l:config = ale#path#FindNearestFile(a:buffer, '.style.yapf')
     let l:config_options = !empty(l:config)
     \   ? ' --style ' . ale#Escape(l:config)
     \   : ''
 
     return {
-    \   'command': 'yapf --no-local-style' . l:config_options,
+    \   'command': l:executable . ' --no-local-style' . l:config_options,
```

Comment:
Apply `ale#Escape` to `l:executable` right here, and in other files. The path to virtualenv might contain special characters.